core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -37,6 +40,11 @@ public AssignUUID(String uuid) {
     public String uuid() {
       return uuid;
     }
+
+    @Override
+    public void applyTo(TableMetadata.Builder metadataBuilder) {
+      throw new UnsupportedOperationException("Not implemented");
```

Comment:
This is not part of the REST API, so I left it unimplemented. I'm not sure whether we will need this because tables have had UUIDs for a long time. It seems unlikely that someone will update from 0.6.0 or earlier directly to 0.14.0 and want to use the REST catalog.